File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return None
        if head.next is None:
            return head

        root = self.reverseList(head.next)
        pointer = root
        while pointer.next is not None:
            pointer = pointer.next
        head.next = None
        pointer.next = head

        return root

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {

    // Get current node larger value
    if list2 == nil {
        return list1
    }
    if list1 == nil {
        return list2
    }
    if list1.Val > list2.Val {
        node := list2
        appendage := mergeTwoLists(list1, list2.Next);
        node.Next = appendage;
        return node;
    } else {
        node := list1
        appendage := mergeTwoLists(list1.Next, list2)
        node.Next = appendage;
        return node;
    }
}
